packages/next/lib/load-custom-routes.ts

```diff
@@ -668,6 +668,23 @@ export default async function loadCustomRoutes(
     }
   }
 
+  const totalRewrites =
+    rewrites.beforeFiles.length +
+    rewrites.afterFiles.length +
+    rewrites.fallback.length
+
+  const totalRoutes = headers.length + redirects.length + totalRewrites
+
+  if (totalRoutes > 1024) {
+    console.warn(
+      `Warning total number of custom routes exceeds 1024, this can reduce performance. Route counts:\n` +
```

Comment:
Should this use chalk like we do with other warnings?

https://github.com/vercel/next.js/blob/8c3c2b7ea2c6ee5cfaeceae1137c2c352bf66dfa/packages/next/build/index.ts#L905-L910